### DISCUSSION TITLE
Having cabal see dependencies brought in in flake

### DISCUSSION THREAD

**Discussion by reinux** [srid/haskell-flake#211] [Q&A] [answered]
I may be setting it up incorrectly to begin with, but I'm importing a Haskell library (also set up using sred/haskell-flake) from git:

```
inputs = {
    xmobar-prusalink.url = "github:reinux/xmonad-prusalink";
};
outputs =
    ...
    packages = {
        xmobar-prusalink.source = inputs.xmobar-prusalink;
    };
};
```

This works fine when I do a `nix build`, but even when I add it to the .cabal under `build-depends`, I get this:
```
Resolving dependencies...
Error: cabal: Could not resolve dependencies:
[__0] trying: xmobar-rei-0.1.0.0 (user goal)
[__1] unknown package: xmobar-prusalink (dependency of xmobar-rei)
[__1] fail (backjumping, conflict set: xmobar-prusalink, xmobar-rei)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: xmobar-rei, xmobar-prusalink
```

This also means that HLS won't see it either.

(For background, I'm experienced with F# and Debian but haven't touched Haskell since college and I'm new to Nix.)

Thanks in advance!

**Comment by roberth** [comment, marked as answer]
I think you're looking for https://github.com/srid/haskell-flake/blob/master/doc/guide/modules.md#default-modules
Make sure your own flake also imports `haskell-flake` and then add something like this to `perSystem`:
```nix
  haskellProjects.default = {
    imports = [
      inputs.xmobar-prusalink.haskellFlakeProjectModules.output
    ];
#...
```

**Reply by srid** [reply]
Link to published docs: https://community.flake.parts/haskell-flake/modules#default-modules

**Reply by reinux** [reply]
Sorry for the late reply. That worked. Thanks!

I should have known it's in the docs. Unfortunately, it's taking me a while to cement a vocabulary and mental model around flakes and nix packages in general.
